src/webassembly.rs

```diff
@@ -85,9 +89,108 @@ pub fn run_instance(
 ) -> CallResult<()> {
     if is_emscripten_module(module) {
         run_emscripten_instance(module, instance, path, args)?;
+
+        Ok(())
     } else {
-        instance.call("main", &[])?;
-    };
+        instance
+            .exports()
+            .find_map(
+                |(export_name, export)| match (export_name.as_ref(), export) {
+                    ("main", Export::Function { signature, .. }) => Some(signature),
+                    _ => None,
+                },
+            )
+            .ok_or_else(|| {
+                CallError::Resolve(ResolveError::ExportNotFound {
+                    name: "main".to_string(),
+                })
+            })
+            .and_then(|signature| {
+                let signature = signature.clone();
+                let parameter_types = signature.params();
+
+                if args.len() != parameter_types.len() {
+                    Err(CallError::Resolve(ResolveError::Signature {
+                        expected: (*signature).clone(),
+                        found: args.iter().map(|_| Type::I32).collect(),
```

Comment:
We don't know the type of the arguments yet, we need to parse them. For the error, let's assume it's `i32`.